### PR TITLE
stats: reduce conflict for updating column size

### LINKED ISSUE #7079: statistics update has too many conflicts
We periodically update `mysql.stats_histogram` table, execute update statements for each column in the table in a  transaction.

If the table has many columns, the transaction will take a long time to commit, the chance of write-conflict is high.

One way to solve the problem is to first select all histogram for a table in `mysql.stats_histogram`, then use `insert on duplicate update statement` to update the rows.

### PR DESCRIPTION
**Pull request by lamxTyler**
## What have you changed? (mandatory)

By using `insert ... on duplicate`, we can reduce the possibility of conflict because of less execution time. 
Fix #7079.
PTAL @coocood @zz-jason @winoros 

## What is the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Existing unit test.

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No.

## Does this PR affect tidb-ansible update? (mandatory)

No.

## Does this PR need to be added to the release notes? (mandatory)

No.

### CODE REVIEW

File: statistics/update.go [hunk 1]
```diff
@@ -343,18 +345,37 @@ func (h *Handle) dumpTableStatColSizeToKV(id int64, delta variable.TableDelta) (
...
 	for key, val := range delta.ColSize {
 		if val == 0 {
 			continue
 		}
-		sql := fmt.Sprintf("update mysql.stats_histograms set version = %d, tot_col_size = tot_col_size + %d where hist_id = %d and table_id = %d and is_index = 0", version, val, key, id)
-		_, err = exec.Execute(ctx, sql)
-		if err != nil {
-			return
-		}
+		values = append(values, fmt.Sprintf("(%d, 0, %d, 0, %d)", id, key, val))
```
Comment (coocood) [inline statistics/update.go]: The `val` is delta, we need to insert the new value.

File: statistics/update.go [hunk 2]
```diff
@@ -343,18 +345,37 @@ func (h *Handle) dumpTableStatColSizeToKV(id int64, delta variable.TableDelta) (
 	defer func() {
 		err = finishTransaction(ctx, exec, err)
 	}()
-	version := h.mu.ctx.Txn().StartTS()
 
+	sql := fmt.Sprintf("select * from mysql.stats_histograms where table_id = %d and is_index = 0 for update", id)
```
Comment (coocood) [inline statistics/update.go]: We should select needed columns instead of `*` for compatibility.

File: statistics/update.go [hunk 3]
```diff
@@ -344,17 +345,16 @@ func (h *Handle) dumpTableStatColSizeToKV(id int64, delta variable.TableDelta) (
 		err = finishTransaction(ctx, exec, err)
 	}()
 	version := h.mu.ctx.Txn().StartTS()
-
+	values := make([]string, 0, len(delta.ColSize))
 	for key, val := range delta.ColSize {
```
Comment (coocood) [inline statistics/update.go]: rename to `histID, deltaColSize` would be more clear.

**Comment by jackysp** [comment]
How to know that it is effective?

**Comment by coocood** [comment]
@lamxTyler we need to do a manual test and verify that the conflict is reduced.

**Pull request merged**